extensions/sql/processors/ExecuteSQL.cpp

```
@@ -0,0 +1,156 @@
+        "If this property is specified, it will be used regardless of the content of incoming flowfiles. "
+        "If this property is empty, the content of the incoming flow file is expected to contain a valid SQL select query, to be issued by the processor to the database. "
+        "Note that Expression Language is not evaluated for flow file contents.")->supportsExpressionLanguage(true)->build());
+
+static core::Property MaxRowsPerFlowFile(
+	core::PropertyBuilder::createProperty("Max Rows Per Flow File")->isRequired(true)->withDefaultValue<int>(0)->withDescription(
+		"The maximum number of result rows that will be included intoi a flow file. If zero then all will be placed into the flow file")->supportsExpressionLanguage(true)->build());
+
+core::Relationship ExecuteSQL::Success("success", "Successfully created FlowFile from SQL query result set.");
+
+ExecuteSQL::ExecuteSQL(const std::string& name, utils::Identifier uuid)
+    : core::Processor(name, uuid), max_rows_(0),
+      logger_(logging::LoggerFactory<ExecuteSQL>::getLogger()) {
+}
+
+ExecuteSQL::~ExecuteSQL() {
+}
+
+void ExecuteSQL::initialize() {
+  //! Set the supported properties
+  setSupportedProperties( { DBCControllerService, s_SQLSelectQuery, MaxRowsPerFlowFile });
+
+  //! Set the supported relationships
+  setSupportedRelationships( { Success });
+}
+
+void ExecuteSQL::onSchedule(const std::shared_ptr<core::ProcessContext> &context, const std::shared_ptr<core::ProcessSessionFactory> &sessionFactory) {
+  context->getProperty(DBCControllerService.getName(), db_controller_service_);
+  context->getProperty(s_SQLSelectQuery.getName(), sqlSelectQuery_);
+  context->getProperty(MaxRowsPerFlowFile.getName(), max_rows_);
+
+  database_service_ = std::dynamic_pointer_cast<sql::controllers::DatabaseService>(context->getControllerService(db_controller_service_));
+  if (database_service_ == nullptr) {
+    logger_->log_error("'DB Controller Service' must be defined");
```

You can throw an exception here, that's going to prevent onTrigger getting called. 

Please take a look at this PR:
https://github.com/apache/nifi-minifi-cpp/pull/683

---

Fixed.